### PR TITLE
Remove pytorch lightning.callbacks.lr monitor.learning rate monitor.lr_sch_names

### PR DESCRIPTION
**Pull request by rschireman**
## What does this PR do?

Fixes part of https://github.com/Lightning-AI/lightning/issues/12521 by removing the deprecated `pytorch lightning.callbacks.lr monitor.learning rate monitor.lr_sch_names`

### Does your PR introduce any breaking changes? If yes, please list them.

Yes, removes pytorch lightning.callbacks.lr monitor.learning rate monitor.lr_sch_names`

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/Lightning-AI/lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by rschireman** [comment]
@akihironitta

**Comment by akihironitta** [comment]
Hi @rschireman, thanks for working on this!

Could you add a changelog entry for both `add_lr_sch_names` and `lr_sch_names`? Once it's added, let's mark this PR as ready for review :]

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
